Tests/Fuzzers/FuzzTarget.cpp

```diff
@@ -1,87 +1,76 @@
-#include <iostream>
-
-#include <IPv4Layer.h>
-#include <Packet.h>
 #include <PcapFileDevice.h>
+#include <Packet.h>
+#include <Logger.h>
+#include "DumpToFile.h"
+#include "ReadParsedPacket.h"
 
-#include "Logger.h"
-
-#define TMP_FILEPATH "/tmp/fuzz_sample.pcap"
+static std::string tmpName;
+static std::string tmpFile;
 
-// This function is created as PcapPlusPlus doesn't seem to offer a way of
-// parsing Pcap files directly from memory
-int dumpDataToPcapFile(const uint8_t *data, size_t size)
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size)
 {
-	FILE *fd;
-	int written = 0;
+	if (tmpName.empty())
+		tmpName = tmpnam (NULL);
 
-	fd = fopen(TMP_FILEPATH, "wb");
-	if (fd == NULL)
-	{
-		std::cerr << "Error opening pcap file for writing\n";
-		return -1;
-	}
+	if (tmpFile.empty())
+		tmpFile = tmpName + FILE_EXT;
 
-	written = fwrite(data, 1, size, fd);
-	if (static_cast<size_t>(written) != size)
+	if (dumpDataToPcapFile(data, size, tmpFile.c_str()) < 0)
 	{
-		std::cerr << "Error writing pcap file\n";
-		fclose(fd);
-		return -1;
+		std::cerr << "Can't Dump buffer to the '" << tmpFile << "' file!!!!\n";
+		return 1;
```

Comment:
If I remember correctly the return value matters, but I don't remember how. Can you please explain when it's recommended to return `0` and when we should return non-zero value like `1`?

Reply:
Good point, I checked the documentation and it looks like anything other than 0 and -1 is reserved. https://llvm.org/docs/LibFuzzer.html#id23 https://llvm.org/docs/LibFuzzer.html#id41
I have pushed a fix.

The rule of thumb I used: if situation is expected (`getNextPackets` fails, but doesn't crash) - return 0. In abnormal situation - return an error code.